### PR TITLE
streamingccl: prioritize cutover on job resumption

### LINKED ISSUE #84008: streamingccl: prioritize requested cutover during resume
Stream ingestion job upon resumption should at all costs avoid anything non-cutover-related get in the way of it being able to perform the cutover.

Jira issue: CRDB-17402
Epic CRDB-10147

### PR DESCRIPTION
**Pull request by samiskin**
Resolves #84008 

During a disaster, cutover is the most important task required to bring
the backup cluster online and ready to receive requests.  Previously if
unrelated errors happened such as a planning error we would be unable to
cutover.  This change checks for a cutover time prior to the highwater
as soon as the job resumes and completes the cutover if required.

Release note (bug fix): streaming cutover no longer relies on successful
job planning

### CODE REVIEW

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go:222]: 💭 Perhaps we can move this out to a top level method. Then maybeRevertToCutoverTimestamp is right at the top of the function, and we can to the comment above calling it that any code between the start of resume and the cutover needs to be looked at very carefully.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go:237]: [nit] missing period here and in a few other comments.

File: pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go:444]: Is it failing with #83694 or something else?  If it is that issue, can we use `UnderRaceWithIssue`

**Review by stevendanna — DISMISSED** [review]

**Comment by samiskin** [comment]
@miretskiy Requesting from you since the others are out today

**Comment by samiskin** [comment]
The test failure is something I fixed in https://github.com/cockroachdb/cockroach/pull/84286 so I'll just wait until that's merged to merged this in

**Review by gh-casper — APPROVED** [review]

**Pull request merged**
